### PR TITLE
Add GetClient method

### PR DESCRIPTION
**Pull request by jfarrell**
Adds a GetClient method to return the httpClient currently set.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -716,6 +716,11 @@ func (c *Client) IsProxySet() bool {
 	return c.proxyURL != nil
 }
 
+// GetClient method returns the current http.Client used by the resty client. 
+func (c *Client) GetClient() *http.Client {
```
Comment (jeevatkm) [inline client.go]: The same `GetClient` method required for default client too at this file [default.go](https://github.com/go-resty/resty/blob/master/default.go). Could you please add it?

```go
// GetClient method returns the current `http.Client` used by the default resty client.
func GetClient() *http.Client {
  return DefaultClient.httpClient
}
```

And adding test coverage for these two methods.

Comment (jfarrell) [reply client.go]: +1, can do

Comment (jeevatkm) [reply client.go]: Thank you @jfarrell

**Comment by jfarrell** [comment]
@jeevatkm updated default.go and added test cases, let me know if you would like anything else for this. Thanks for reviewing

**Comment by jeevatkm** [comment]
@jfarrell Thank you.

**Pull request merged**